### DISCUSSION TITLE
Problem to install firmware with crazyflie clien.

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#1186] [Q&A]
Hi.

I currently installed crazyflie client again and the following problem arose, when I select the firmware in cf2 and press the program button, the bar does not load and the crazyflie immediately leaves the bootloader mode and turns on, I have investigated if anyone else has had the same problem, but I have not found any information related to this problem, so I would appreciate your help.

Regards

**Comment by ataffanel** [comment]
Hi,

This is not something that I have seen before, we would need a bit more information to be able to investigate what the problem might be. What version of the client are you using and what OS are you running on your computer?

**Reply by Wallat1** [reply]
Hi.

Thanks for your help. I am using the latest version with ubuntu 20.04. I must add, that I installed it on another laptop with the same OS and the problem continues.

Regards.

**Reply by knmcguire** [reply]
@Wallat1 Do you know which firmware is running on the crazyflie before you try to flash it? Also, from the terminal that you have run the cfclient, do you see any error when clicking the program button?

**Reply by Wallat1** [reply]
Hi.

Thanks for your help, the firmware was 2020.06 cf, at first glance it doesn't seem like there is any error, however I'm going to check it carefully. On the other hand, on a desktop PC I have an older version of cfclien and I have no problems updating the firmware, for this reason I think the problem is in the current version.

Regards.

**Reply by ataffanel** [reply]
We just updated the client and libs a couple of days ago, could you check that you are actually running the very latest version of both? This can be done by running ```pip install --upgrade cfclient cflib```

**Reply by Wallat1** [reply]
Hi.

I appreciate your help, I have run pip --upgrade cfclient cflib and cfclien already allows me to update the firmware, but only to version 2024 and not to previous ones, so the problem has only been partially resolved.

Regards.

**Reply by ataffanel** [reply]
Thanks for testing, this does seem to be a bug, I created an issue on the client repos and we will start looking at it: https://github.com/bitcraze/crazyflie-clients-python/issues/702. I will try to keep this thread updated when we fix it but you should also follow the issue to make sure you are updated.

In the mean time, you can always install the old client/lib with ```pip uninstall --purge cfclient cflib; pip install cfclient==2023.11```.

**Reply by evoggy** [reply]
Could you please try to run the client from the command line and then try to upgrade/downgrade again and past the text from the terminal? This will give us some more clues as to what might be going wrong.

**Reply by Wallat1** [reply]
Hi.

I have done what I was told and the problem still persists with the new version of cfclient, it will only allow me to update to version 2024.2, but now that the Crazyflie has been updated to version 2024.2, the Crazyflie does not fly correctly and does not allow me to return to a previous version with the cfclient that I have on another PC, since when the update is finished only the blue M2 LED turns on, I don't know if I am the only person with this problem or if there is a problem, I would greatly appreciate your help.

The version of cfclient that worked for me until yesterday is 2022.3.post20+63f0516.

Regards.

**Comment by Wallat1** [comment]
Hi.

@evoggy , when I run the cfclient, this appears:
![1](https://github.com/bitcraze/discussions/assets/112857979/06021fa2-11dc-42e8-8f6c-85a521516be8)
![2](https://github.com/bitcraze/discussions/assets/112857979/c1e83e06-8700-4610-82ef-e278f6c9413e)

when upgrading to the version 2024.2 cf this appears:
![3](https://github.com/bitcraze/discussions/assets/112857979/fd2680ab-78f9-4074-985f-cdb98e5617bf)

and when I select older versions this happens:
![Captura de pantalla de 2024-02-27 16-24-05](https://github.com/bitcraze/discussions/assets/112857979/8838eea5-f589-4cd2-ac8f-e1e98a8096eb)

I hope this can be helpful.

Regards.

**Reply by evoggy** [reply]
Thanks for the screenshots. Just so I understand, does the Crazyflie start up correctly in the cases you have screenshots for above?

**Reply by Wallat1** [reply]
Hi.

Thanks for responding, it starts fine when version 2024.2 cf is loaded, and it restarts for any of the other versions, I think that through screenshots it is very difficult to illustrate what is happening, so I am going to share some videos showing what is happening, I will do this as soon as possible, since I do not have a crazyflie with me at the moment.

Regards.

**Reply by Wallat1** [reply]
Hi.

@evoggy , I hope this video to be useful.

https://github.com/bitcraze/discussions/assets/112857979/72a5b5d6-3891-419d-bddb-91dcb456dece



Regards.

**Reply by evoggy** [reply]
Thanks for the video!

I've tried downgrading from 2024.02 to 2020.06 and I cannot reproduce the issue you're seeing. When I look at the video there's no printouts when you press program in the console, is this correct? Is this the same Crazyflie you can flash 2024.02 and it works? Can you flash any other version, like 2023.11, and it's working?

For the second part it seems as if the controller input is bad, look at the image below. The controller output is set to -30 degrees and the motor output shows that it's flying sharply to the right. Try to check so that roll/pitch inputs are around 0 before trying to fly again and see if it's better.

![30-deg-pitch](https://github.com/bitcraze/discussions/assets/5947128/40d3ccc6-3c07-4315-b38e-3d01d7934ea2)

**Reply by Wallat1** [reply]
Hi.

Thanks for the response, when I watch the video there are no impressions when you press program on the console, is this correct? , yes, it is correct, I have that problem on 3 PCs with Ubuntu 20.04 and two with W10, regarding the last video, this one the problem only  occurs on two PCs with Ubuntu, if there is no problem with cfclient, I don't know what I did wrong when installing cfclient, I have followed the instructions step by step as shown on the git hub page.

Is this the same Crazyflie you can flash 2024.02 and it works?, yes.

Can you flash any other version, like 2023.11, and it's working?, no, I can only flash to  version 2024.02 as shown in the second video.

Regards.

**Reply by Wallat1** [reply]
Hi.

Thank you for your attention, I can solve the problem, the problem was related to python version 3.8 and 3.9, for this reason, I updated python to the version 3.12.

Regards.

**Reply by knmcguire** [reply]
Thanks for notifying us! It would have been able to work with 3.8 and up but we will double check this. I've just made a ticket:  https://github.com/bitcraze/crazyflie-clients-python/issues/704
